tests/rados/test_cephdf.py

```diff
@@ -286,3 +293,185 @@ def run(ceph_cluster, **kw):
+
+            _pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+            log.info(f"{pool_name} pool stat: {_pool_stat}")
+
+            time.sleep(30)
+            # execute max_avail check across the cluster
+            if not rados_obj.verify_max_avail():
+                log.error("MAX_AVAIL deviates on the cluster more than expected")
+            log.info("MAX_AVAIL on the cluster are as per expectation")
+
+            # expand the OSD LVMs on backup node12
+            node12_obj = get_node_by_id(ceph_cluster, "node12")
+            for lvm in lvm_list["node12"]:
+                _cmd = f"lvextend -L 15G {lvm}"
+                node12_obj.exec_command(cmd=_cmd, sudo=True)
+
+            log.info("LVM size on node12 increased from 10G to 15G")
+            node12_osds = rados_obj.collect_osd_daemon_ids(osd_node=node12_obj)
+
+            # for each OSD on node 12, execute bluefs-bdev-expand
+            for osd_id in node12_osds:
+                out = bluestore_obj.block_device_expand(osd_id=osd_id)
+                log.info(out)
+                assert "device size" in out and "Expanding" in out
+            log.info(f"OSDs {node12_osds} should now be 15G each")
+
+            _pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+            log.info(f"{pool_name} pool stat: {_pool_stat}")
+
+            time.sleep(30)
+            # execute max_avail check across the cluster
+            if not rados_obj.verify_max_avail():
+                log.error("MAX_AVAIL deviates on the cluster more than expected")
+            log.info("MAX_AVAIL on the cluster are as per expectation")
+
+            # write data to the pool and expand OSD LVMs on backup node13
+            assert rados_obj.bench_write(pool_name=pool_name, rados_write_duration=300)
+
+            _pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+            log.info(f"{pool_name} pool stat: {_pool_stat}")
+
+            time.sleep(30)
+            # execute max_avail check across the cluster
+            if not rados_obj.verify_max_avail():
+                log.error("MAX_AVAIL deviates on the cluster more than expected")
+            log.info("MAX_AVAIL on the cluster are as per expectation")
+
+            # expand the OSD LVMs on backup node13
+            node13_obj = get_node_by_id(ceph_cluster, "node13")
+            for lvm in lvm_list["node13"]:
+                _cmd = f"lvextend -L 13G {lvm}"
+                node13_obj.exec_command(cmd=_cmd, sudo=True)
+
+            log.info("LVM size on node13 increased from 10G to 13G")
+            node13_osds = rados_obj.collect_osd_daemon_ids(osd_node=node13_obj)
+
+            # for each OSD on node 12, execute bluefs-bdev-expand
+            for osd_id in node13_osds:
+                out = bluestore_obj.block_device_expand(osd_id=osd_id)
+                log.info(out)
+                assert "device size" in out and "Expanding" in out
+
```

Comment:
Check osd metadata if the increased size is reported correctly post bdev expand workflow.

Reply:
Added code to verify bluestore_bdev_size from OSD metadata